app/models/challenge_models/gift_exchange.rb

```diff
@@ -80,4 +80,21 @@ def user_allowed_to_see_requests_summary?(user)
   def user_allowed_to_see_prompt?(user, prompt)
     self.collection.user_is_maintainer?(user) || prompt.pseud.user == user
   end
+
+  # reindex collection after creation, deletion, and certain attribute updates
+  after_create :reindex_collection
+  after_destroy :reindex_collection
+  after_update :reindex_collection, if: Proc.new { |c| c.saved_change_to_signup_open? ||
```

Comment:
You should use `after_commit` for the new reindexing callbacks if possible. We plan to update the existing ones as well ([AO3-5944](https://otwarchive.atlassian.net/browse/AO3-5944)).